packages/react/label/src/Label.tsx

```diff
@@ -22,37 +25,21 @@ interface LabelProps extends PrimitiveSpanProps {
 
 const Label = React.forwardRef<LabelElement, LabelProps>((props, forwardedRef) => {
   const { htmlFor, id: idProp, ...labelProps } = props;
-  const labelRef = React.useRef<HTMLSpanElement>(null);
-  const ref = useComposedRefs(forwardedRef, labelRef);
+  const [control, setControl] = React.useState<HTMLElement | null>(null);
+  const ref = React.useRef<HTMLSpanElement>(null);
+  const composedRefs = useComposedRefs(forwardedRef, ref);
   const id = useId(idProp);
 
-  React.useEffect(() => {
-    const label = labelRef.current;
-
-    if (label) {
-      const handleMouseDown = (event: MouseEvent) => {
-        if (event.detail > 1) event.preventDefault();
-      };
-
-      // prevent text selection when double clicking label
-      label.addEventListener('mousedown', handleMouseDown);
-      return () => label.removeEventListener('mousedown', handleMouseDown);
-    }
-  }, [labelRef]);
-
   React.useEffect(() => {
     if (htmlFor) {
       const element = document.getElementById(htmlFor);
-      const label = labelRef.current;
-
+      const label = ref.current;
       if (label && element) {
-        const removeLabelClickEventListener = addLabelClickEventListener(label, element);
         const getAriaLabel = () => element.getAttribute('aria-labelledby');
         const ariaLabelledBy = [getAriaLabel(), id].filter(Boolean).join(' ');
         element.setAttribute('aria-labelledby', ariaLabelledBy);
-
+        setControl(element);
```

Comment:
Does this need to be state? or would a `ref` work?

Reply:
Oooh good shout, changed!